### PR TITLE
feat(server): add queryDatasetByName impl

### PR DESCRIPTION
**Pull request by fenngZ**
feat(server): add queryDatasetByName impl

### CODE REVIEW

File: server/src/main/java/io/openbytes/odi/interfaces/ro/ListDatasetRO.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+public class ListDatasetRO {
+
+    // page
+    private Integer index;
+
+    private Integer size;
+
+    private Map<String, Boolean> orderFields;
+
+    private String keyWord;
```
Comment (ziyiyizi) [inline server/src/main/java/io/openbytes/odi/interfaces/ro/ListDatasetRO.java:33]: rename

File: server/src/main/java/io/openbytes/odi/domain/repository/DatasetRepository.java [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 package io.openbytes.odi.domain.repository;
 
+import com.baomidou.mybatisplus.core.metadata.IPage;
+import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
```
Comment (ziyiyizi) [inline server/src/main/java/io/openbytes/odi/domain/repository/DatasetRepository.java:20]: Should there be no dependency on implementation classes in the interface layer？

**Review by ziyiyizi — CHANGES_REQUESTED** [review]
.

File: server/src/main/java/io/openbytes/odi/domain/repository/DatasetRepository.java (hunk 2 above)
Comment (fenngZ) [reply server/src/main/java/io/openbytes/odi/domain/repository/DatasetRepository.java:19]: yes and done.

File: server/src/main/java/io/openbytes/odi/interfaces/ro/ListDatasetRO.java (hunk 1 above)
Comment (fenngZ) [reply server/src/main/java/io/openbytes/odi/interfaces/ro/ListDatasetRO.java]: done

File: server/src/main/java/io/openbytes/odi/domain/repository/DatasetRepository.java (hunk 2 above)
Comment (ziyiyizi) [reply server/src/main/java/io/openbytes/odi/domain/repository/DatasetRepository.java:19]: 👿

**Review by ziyiyizi — APPROVED** [review]

**Pull request merged**